docs/howto/code/flask-async/task.yaml

```diff
@@ -0,0 +1,194 @@
+
+  # Install the latest rockcraft snap
+  # (This can be removed after the Rockcraft PR is merged)
+  # The PR: https://github.com/canonical/rockcraft/pull/747
+  snap install snapcraft --channel=latest/edge --classic
+  # Download rockcraft async-workers branch and alithethird fork
+  git clone -b flask-django-extention-async-workers https://github.com/alithethird/rockcraft
+  cd rockcraft
+  snapcraft pack
+  snap install --dangerous --classic rockcraft_*
+
+  # snap refresh charmcraft --channel=latest/edge --amend
+  snap install microk8s --channel=1.31-strict/stable
+  snap install juju --channel=3/stable
+
+  mkdir -p ~/.local/share
+
+  # MicroK8s config setup
+  microk8s status --wait-ready
+  microk8s enable hostpath-storage
+  microk8s enable registry
+  microk8s enable ingress
+
+  # Bootstrap controller
+  juju bootstrap microk8s dev-controller
+
+  cd $HOME
+  # [docs:create-venv]
+  sudo apt-get update && sudo apt-get install python3-venv -y
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install -r requirements.txt
+  # [docs:create-venv-end]
+
+  flask run -p 8000 &
+  retry -n 5 --wait 2 curl --fail localhost:8000
+
+  # [docs:curl-flask]
+  curl localhost:8000
+  # [docs:curl-flask-end]
+
+  # [docs:curl-flask-async-app]
+  curl localhost:8000/io
+  # [docs:curl-flask-async-app-end]
+
+  kill $!
+
+  # [docs:create-rockcraft-yaml]
+  rockcraft init --profile flask-framework
+  # [docs:create-rockcraft-yaml-end]
+
+  sed -i "s/name: .*/name: flask-async-app/g" rockcraft.yaml
+  sed -i "s/amd64/$(dpkg --print-architecture)/g" rockcraft.yaml
+
+  # [docs:pack]
+  rockcraft pack
+  # [docs:pack-end]
+
+  # [docs:ls-rock]
+  ls *.rock -l
+  # [docs:ls-rock-end]
+
+  # [docs:skopeo-copy]
+  rockcraft.skopeo --insecure-policy copy --dest-tls-verify=false \
+    oci-archive:flask-async-app_0.1_$(dpkg --print-architecture).rock \
+    docker://localhost:32000/flask-async-app:0.1
+  # [docs:skopeo-copy-end]
+
+  # [docs:create-charm-dir]
+  mkdir charm
+  cd charm
+  # [docs:create-charm-dir-end]
+
+  # [docs:charm-init]
+  charmcraft init --profile flask-framework --name flask-async-app
+  # [docs:charm-init-end]
+
+  sed -i "s/paas-charm.*/https:\/\/github.com\/canonical\/paas-charm\/archive\/async-workers.tar.gz/g" requirements.txt
+
+  # [docs:charm-pack]
+  charmcraft pack
+  # [docs:charm-pack-end]
+
+  # [docs:ls-charm]
+  ls *.charm -l
+  # [docs:ls-charm-end]
+
+  # [docs:add-juju-model]
+  juju add-model flask-async-app
+  # [docs:add-juju-model-end]
+
+  juju set-model-constraints -m flask-async-app arch=$(dpkg --print-architecture)
+
+  # [docs:deploy-juju-model]
+  juju deploy ./flask-async-app_ubuntu-22.04-$(dpkg --print-architecture).charm \
+    flask-async-app --resource \
+    flask-app-image=localhost:32000/flask-async-app:0.1
+  # [docs:deploy-juju-model-end]
+
+  # [docs:deploy-nginx]
+  juju deploy nginx-ingress-integrator --channel=latest/edge --revision 122
```

Comment:
You may want to specify the `--base` instead of the `--revision` here, unless there's a specific reason why you opted for revision

Reply:
Aligned to your PR. Thank you!